server/src/test/java/io/crate/lucene/NumericEqQueryTest.java

```diff
@@ -68,4 +71,19 @@ public void test_uses_binary_encoded_range_queries_for_large_numeric() throws Ex
         assertThat(query).isInstanceOf(PointRangeQuery.class);
         assertThat(query.toString()).isEqualTo("y:[-99999999999999999999999999999999999999 TO 274679983711617676]");
     }
+
+    @Test
+    public void test_equals_with_different_precision() {
```

Comment:
Only `x = 1.111::numeric` fails, the other asserts are added for completeness. Without the fix the query becomes `x:[1111 TO 1111]` which is equivalent to `x = 11.11::numeric`.